src/containers/SecretsModal/SecretsModal.js

```diff
@@ -253,12 +253,14 @@ export /* istanbul ignore next */ class SecretsModal extends Component {
         newInvalidFields.push(id);
       }
 
-      id = id
-        .split('-')
-        .splice(1, 2)
-        .join();
-      const index = id.substring(id.length - 1);
-      id = id.slice(0, -1);
+      let index;
```

Comment:
I don't follow what's happening here or how this addresses the issue. It could probably benefit from some comments, maybe including expected format of `id` etc.

We should avoid string manipulation to extract useful data as much as possible, preferring instead to pass objects containing the relevant fields.

Reply:
done!